### PR TITLE
haskellPackages.hadolint: unbreak

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/haskell-modules/configuration-common.nix [hunk 1]
```diff
@@ -1715,6 +1715,9 @@ self: super: {
   # Issue reported upstream, no bug tracker url yet.
   darcs = doJailbreak super.darcs;
 
+  # Too strict version bounds on ansi-terminal;
+  colourista = doJailbreak super.colourista;
```
Comment (cdepillabout) [inline pkgs/development/haskell-modules/configuration-common.nix:1719]: Could you report this upstream, and link the issue here?

Comment (SuperSandro2000) [reply pkgs/development/haskell-modules/configuration-common.nix:1719]: There is already an upstream commit that fixes the issue https://github.com/kowainik/colourista/commit/15ace92105b56eba4ea3717bd55f733afe5be401. Not sure how to get that into package though.

Comment (sternenseemann) [reply pkgs/development/haskell-modules/configuration-common.nix:1719]: You can either use `appendPatch` or just add a “remove at release after <current release>” to the comment.

Comment (sternenseemann) [reply pkgs/development/haskell-modules/configuration-common.nix:1719]: Example:

<https://github.com/NixOS/nixpkgs/blob/0c9f9676c7ccadde3cd05f399898bbfe7d5a0831/pkgs/development/haskell-modules/configuration-common.nix#L713-L718>

Comment (SuperSandro2000) [reply pkgs/development/haskell-modules/configuration-common.nix]: Converted it to the patch.

File: pkgs/development/haskell-modules/configuration-common.nix [hunk 2]
```diff
@@ -1715,6 +1715,13 @@ self: super: {
   # Issue reported upstream, no bug tracker url yet.
   darcs = doJailbreak super.darcs;
 
+  # Too strict version bounds on ansi-terminal;
+  colourista = doJailbreak (appendPatch super.colourista
```
Comment (sternenseemann) [inline pkgs/development/haskell-modules/configuration-common.nix:1719]: With the patch, jailbreak shouldn't be necessary (or rather the patch is kinda pointless).

**Review by sternenseemann — CHANGES_REQUESTED** [review]

**Comment by sternenseemann** [comment]
Thanks!

**Comment by SuperSandro2000** [comment]
@sternenseemann thanks! The first time I touched any haskell.

**Comment by sternenseemann** [comment]
Hope the experience doesn't deter you in the future :) It's a bit unique, I really should translate the existing documentation on the process to written form soon™.

**Comment by cdepillabout** [comment]
@SuperSandro2000 Yeah, thanks for taking the time to do this update :-)

**Pull request merged**